service/k8sclient/k8sclient.go

```diff
@@ -664,13 +697,14 @@ func (c *K8sClient) RestartXtraDBCluster(ctx context.Context, name string) error
 		return err
 	}
 
-	// TODO: implement logic to handle the case when there is HAProxy instead of ProxySQL.
-	_, err = c.kubeCtl.Run(ctx, c.restartDBClusterCmd(name, "proxysql"), nil)
-	if err != nil {
-		return err
+	for _, proxy := range []string{"proxysql", "haproxy"} {
+		if _, err := c.kubeCtl.Run(ctx, []string{"get", "statefulset", name + "-" + proxy}, nil); err == nil {
```

Comment:
will this command fail if there is no proxysql or haproxy?

Reply:
You mean no proxy at all? Then the one gets `return errors.New("failed to restart pxc cluster proxy statefulset")`.

When there is only HAProxy it fails with not found for ProxySQL and vice versa.

Reply:
ok, good